actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -30,6 +30,14 @@ module Renderers
     included do
       class_attribute :_renderers, default: Set.new.freeze
       class_attribute :escape_json_responses, instance_accessor: false, default: true
+
+      redefine_method :escape_json_responses= do |value|
+        ActionController.deprecator.warn(<<~MSG.squish)
+          Setting action_controller.escape_json_responses = true is deprecated and will have no effect in Rails 8.2.
```

Comment:
I think you meant to check if `value` is true?

Reply:
Good catch, fixed in [a12a15e](https://github.com/rails/rails/pull/54840/commits/a12a15ef90faf4a96f24bb9bca2ad89204267e82).